pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,345 @@
+		return false
+	}
+
+	if cr.Spec.ForProvider.DeliveryOptions != nil && resp.DeliveryOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.SendingPoolName) != awsclient.StringValue(resp.DeliveryOptions.SendingPoolName) {
+			return false
+		}
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.TLSPolicy) != awsclient.StringValue(resp.DeliveryOptions.TlsPolicy) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateReputationOptions checks if ReputationOptions Object are up to date
+func isUpToDateReputationOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove ReputationOptions spec and expect an update
+	if cr.Spec.ForProvider.ReputationOptions == nil && resp.ReputationOptions != nil {
+		cr.Spec.ForProvider.ReputationOptions = &svcapitypes.ReputationOptions{}
+		return false
+	}
+
+	if cr.Spec.ForProvider.ReputationOptions != nil && resp.ReputationOptions != nil {
+		if awsclient.BoolValue(cr.Spec.ForProvider.ReputationOptions.ReputationMetricsEnabled) != awsclient.BoolValue(resp.ReputationOptions.ReputationMetricsEnabled) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateTrackingOptions checks if TrackingOptions Object are up to date
+func isUpToDateTrackingOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove TrackingOptions spec and expect an update
+	if cr.Spec.ForProvider.TrackingOptions == nil && resp.TrackingOptions != nil {
+		cr.Spec.ForProvider.TrackingOptions = &svcapitypes.TrackingOptions{}
+		return false
+	}
+
+	// Once disabled, output response will not populate this option anymore
+	if cr.Spec.ForProvider.TrackingOptions != nil && resp.TrackingOptions == nil {
+		return false
+	}
+
+	if cr.Spec.ForProvider.TrackingOptions != nil && resp.TrackingOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.TrackingOptions.CustomRedirectDomain) != awsclient.StringValue(resp.TrackingOptions.CustomRedirectDomain) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateSuppressionOptions checks if SuppressionOptions Object are up to date
+func isUpToDateSuppressionOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove SuppressionOptions from spec and expect an update
+	if cr.Spec.ForProvider.SuppressionOptions == nil && resp.SuppressionOptions != nil {
+		// PutConfigurationSetSuppressionOptions requires empty slice to toggle update
+		cr.Spec.ForProvider.SuppressionOptions = &svcapitypes.SuppressionOptions{SuppressedReasons: []*string{}}
+		return false
+	}
+
+	crSuppressedReasons := make([]*string, 0)
+	awsSuppressedReasons := make([]*string, 0)
+
+	if cr.Spec.ForProvider.SuppressionOptions != nil && cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons != nil {
+		crSuppressedReasons = append(crSuppressedReasons, cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons...)
```

Comment:
`cr` should not be changed during `isUpTodate`.

Reply:
have fixed this issue